lib/smart_proxy_salt/api_request.rb

```diff
@@ -1,44 +1,49 @@
+# frozen_string_literal: true
+
 require 'json'
 require 'net/http'
 require 'net/https'
 require 'uri'
 
-module Proxy::Salt
-  class ApiError < RuntimeError; end
-
-  class ApiRequest
-    attr_reader :url, :username, :password, :auth
-
-    def initialize
-      @url = Proxy::Salt::Plugin.settings.api_url
-      @auth = Proxy::Salt::Plugin.settings.api_auth
-      @username = Proxy::Salt::Plugin.settings.api_username
-      @password = Proxy::Salt::Plugin.settings.api_password
-
-      begin
-        URI.parse(url)
-      rescue URI::InvalidURIError => e
-        raise ConfigurationError.new("Invalid Salt api_url setting: #{e}")
+module Proxy
+  module Salt
+    class ApiError < RuntimeError; end
+    class ConfigurationError < RuntimeError; end
+
+    # SaltStack's Rest API
+    class ApiRequest
+      attr_reader :url, :username, :password, :auth
+
+      def initialize
+        @url = Proxy::Salt::Plugin.settings.api_url
+        @auth = Proxy::Salt::Plugin.settings.api_auth
+        @username = Proxy::Salt::Plugin.settings.api_username
+        @password = Proxy::Salt::Plugin.settings.api_password
+
+        begin
+          URI.parse(url)
+        rescue URI::InvalidURIError => e
+          raise ConfigurationError.new("Invalid Salt api_url setting: #{e}")
+        end
       end
-    end
 
-    def post(path, options = {})
-      uri              = URI.parse(url)
-      http             = Net::HTTP.new(uri.host, uri.port)
-      http.use_ssl     = uri.scheme == 'https'
-      http.verify_mode = OpenSSL::SSL::VERIFY_NONE
-      path = [uri.path, path].join('/') unless uri.path.empty?
+      def post(path, options = {})
+        uri              = URI.parse(url)
+        http             = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl     = uri.scheme == 'https'
+        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+        path = [uri.path, path].join unless uri.path.empty?
```

Comment:
Any reason this no longer joins using slashes?

Reply:
the join('/') was replaced with join() because at the only usage of this method (rest.rb), the method is called with /run which would result in two slashes ('//run'). see https://github.com/theforeman/smart_proxy_salt/blob/14a3dcb96f5174348c8e9c179954f7c6a40056a4/lib/smart_proxy_salt/rest.rb#L10